solidity/contracts/ERC20WithData.sol

```diff
@@ -9,26 +9,25 @@ import '@openzeppelin/contracts/utils/Context.sol';
     @dev Mintable+burnable form of ERC20 with data event support.
 */
 contract ERC20WithData is Context, ERC20 {
-    constructor(string memory name, string memory symbol) public ERC20(name, symbol) {}
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
 
     function mintWithData(
         address to,
         uint256 amount,
         bytes calldata data
-    ) external virtual {
+    ) public {
         _mint(to, amount);
     }
 
     function transferWithData(
-        address from,
         address to,
         uint256 amount,
         bytes calldata data
-    ) external virtual {
-        _transfer(from, to, amount);
+    ) public {
+        _transfer(_msgSender(), to, amount);
     }
 
-    function burnWithData(address from, uint256 amount, bytes calldata data) external virtual {
-        _burn(from, amount);
```

Comment:
Personally I would lean toward passing `from` here but requiring it to match `_msgSender()`. Let the logic be in the contract so that it's easier for users to replace the contract without needing to alter the connector to have their own burn policy.